jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/Repository.java

```diff
@@ -190,4 +192,10 @@ boolean copyAsset(String uniqueId,
     boolean moveAsset(String uniqueId,
                       String location,
                       String name);
+
+    /**
+     * Sets the servlet context to the repository
+     * @param servletContext
+     */
+    void setServletContext(ServletContext servletContext);
```

Comment:
I would ask you to remove this method form the interface, because just `VFSRepository` needs a `ServletContext`. The other implementation `GuvnorRepository` doesn't need this set method. The you can use constructor DI in the `VFSRepository` like shown below:
```java
@Inject
    public VFSRepository(@Named("ioStrategy") IOService ioService, ServletContext servletContext) {
        this.ioService = ioService;
        this.servletContext = servletContext;
    }
```